### PR TITLE
Selected camera bug fix

### PR DESCRIPTION
**Pull request by mgamis-msft**
# What

SDK initializes cameras with one dummy camera:
`{ id: 'camera:id', name: '', deviceType: 'USBCamera' }`
So selectedCamera will have this value { id: 'camera:id', name: '', deviceType: 'USBCamera' } before new cameras are updated. So it makes sense that we only reset selectedCamera to the first camera if we don't see it in the new list of cameras.

Video where I also test adding a new camera and showing that selectedCamera is NOT set again because it is still on the list.
https://user-images.githubusercontent.com/79475487/122629625-49a07100-d073-11eb-928d-4b4778f20584.mp4

# Why

https://skype.visualstudio.com/SPOOL/_workitems/edit/2475044

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/calling-stateful-client/src/CallContext.ts [hunk 1]
```diff
@@ -570,7 +570,7 @@ export class CallContext {
   public setDeviceManagerCameras(cameras: VideoDeviceInfo[]): void {
     this.setState(
       produce(this._state, (draft: CallClientState) => {
-        if ((!draft.deviceManager.cameras || draft.deviceManager.cameras.length === 0) && cameras.length > 0) {
+        if (cameras.length > 0 && !cameras.some((camera) => camera.id === draft.deviceManager.selectedCamera?.id)) {
```
Comment (prprabhu-ms) [inline packages/calling-stateful-client/src/CallContext.ts:573]: Please add comment explaining why this is done in the code. Essentially, copy the PR description into the code comment.

The SDK behaviour is highly unusual, and will definitely trip me up the next time I read this :/

Comment (prprabhu-ms) [reply packages/calling-stateful-client/src/CallContext.ts:573]: Have we asked the SDK team why we get a dummy camera in the first place? That doesn't sound right.

File: packages/calling-stateful-client/src/CallContext.ts [hunk 2]
```diff
@@ -570,7 +570,13 @@ export class CallContext {
   public setDeviceManagerCameras(cameras: VideoDeviceInfo[]): void {
     this.setState(
       produce(this._state, (draft: CallClientState) => {
-        if ((!draft.deviceManager.cameras || draft.deviceManager.cameras.length === 0) && cameras.length > 0) {
+        if (
+          /** SDK initializes cameras with one dummy camera with value { id: 'camera:id', name: '', deviceType: 'USBCamera' } before camera permissions are granted.
```
Comment (alkwa-msft) [inline packages/calling-stateful-client/src/CallContext.ts:574]: can we keep this comment above the if block so its all together?

**Review by alkwa-msft — APPROVED** [review]

File: packages/calling-stateful-client/src/CallContext.ts (hunk 2 above)
Comment (mgamis-msft) [reply packages/calling-stateful-client/src/CallContext.ts]: Sure.

File: packages/calling-stateful-client/src/CallContext.ts (hunk 1 above)
Comment (mgamis-msft) [reply packages/calling-stateful-client/src/CallContext.ts:576]: Sure added a comment and I just let them know about this potential bug here https://teams.microsoft.com/l/message/19:ea8322f638f34acdb74162cacd55c92b@thread.skype/1624308079847?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e9c1fc3-39df-4486-a26a-456d80e80f82&parentMessageId=1624308079847&teamName=Azure%20Communication%20Services&channelName=Calling&createdTime=1624308079847

**Review by prprabhu-ms — APPROVED** [review]

**Review by JamesBurnside — APPROVED** [review]

**Pull request merged**